src/main/scala/util/Annotations.scala

```diff
@@ -104,22 +106,24 @@ case class ResetVectorAnnotation(target: Named, resetVec: BigInt) extends Single
 /** Helper object containing methods for applying annotations to targets */
 object Annotated {
 
-  def srams(
-    component: InstanceId,
+  def srams[T <: Data](
+    component: SyncReadMem[T],
     name: String,
     address_width: Int,
     data_width: Int,
     depth: BigInt,
     description: String,
-    write_mask_granularity: Int): Unit = {
+    write_mask_granularity: Int,
+    uid: () => Int = () => 0): Unit = {
```

Comment:
I would change this from `() => Int` to just `Int`. It doesn't matter that much since it's just an argument to a function, but we don't need to delay creation of the the id anymore and the order of id allocation will be the same whether we do it immediately here or delay it until the Firrtl Annotation is created.

Reply:
done!